Topics/System_Design/Intro_To_Microservices/Intro_To_Microservices.md

```diff
@@ -0,0 +1,165 @@
+# Introduction to Microservices Architecture
+
+## Table of Contents
+
+1. [Overview of Microservices](#overview)
+2. [Advantages and Disadvantages](#adv-dis)
+4. [Principles and Characteristics](#principles)
+5. [Design Considerations](#design)
+6. [Tools and Technologies](#common-tech)
+7. [Best Practices](#best-practices)
+8. [Real-World Examples](#examples)
+
+
+
+## 1. Overview of Microservices <a name="overview"></a>
+
+In the realm of CSC301H1: Introduction to Software Engineering, understanding modern software architecture is crucial. Microservices architecture, a cornerstone of contemporary software development, is a concept that aligns seamlessly with the principles taught in this course.
+
+Microservices architecture presents a paradigm shift in how software systems are designed and developed. Unlike traditional monolithic architectures, where applications are built as a single, tightly integrated unit, microservices architecture emphasizes breaking down complex systems into smaller, independently deployable services.
+
+For CSC301 students, grasping the significance of microservices architecture entails understanding its foundational principles and how they apply to real-world software engineering practices. Microservices enable teams to work more efficiently by allowing them to focus on developing and maintaining small, autonomous services. This aligns with the agile development methods taught in CSC301, where adaptability, collaboration, and rapid iteration are paramount.
+
+In the context of CSC301, microservices architecture offers several advantages. It promotes scalability, allowing teams to scale individual services based on demand, a concept directly relevant to discussions on estimation and prioritization. Moreover, the autonomy of microservices facilitates teamwork skills, as teams can work independently on different services while ensuring seamless integration through well-defined APIs.
```

Comment:
What does the word "estimation" mean for individual services?

Reply:
I will add a one line definition for the given context